Build failure: datalad

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *datalad*

### Build log

<details>
<summary>
Relevant part:
</summary>

```
=================================== FAILURES ===================================
__________________________ test_subsuperdataset_save ___________________________

path = '/build/datalad_temp_test_subsuperdataset_save2z5t3rub'

    @with_tempfile(mkdir=True)
    def test_subsuperdataset_save(path=None):
        # Verify that when invoked without recursion save does not
        # cause querying of subdatasets of the subdataset
        # see https://github.com/datalad/datalad/issues/4523
        parent = Dataset(path).create()
        # Create 3 levels of subdatasets so later to check operation
        # with or without --dataset being specified
        sub1 = parent.create('sub1')
        sub2 = parent.create(sub1.pathobj / 'sub2')
        sub3 = parent.create(sub2.pathobj / 'sub3')
        assert_repo_status(path)
        # now we will lobotomize that sub3 so git would fail if any query is performed.
        (sub3.pathobj / '.git' / 'config').chmod(0o000)
        try:
            sub3.repo.call_git(['ls-files'], read_only=True)
            raise SkipTest
        except CommandError:
            # desired outcome
            pass
        # the call should proceed fine since neither should care about sub3
        # default is no recursion
        parent.save('sub1')
        sub1.save('sub2')
        assert_raises(CommandError, parent.save, 'sub1', recursive=True)
        # and should not fail in the top level superdataset
        with chpwd(parent.path):
            save('sub1')
        # or in a subdataset above the problematic one
        with chpwd(sub1.path):
>           save('sub2')

datalad/core/local/tests/test_save.py:278: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
datalad/interface/base.py:773: in eval_func
    return return_func(*args, **kwargs)
datalad/interface/base.py:763: in return_func
    results = list(results)
datalad/interface/base.py:873: in _execute_command_
    for r in _process_results(
datalad/interface/utils.py:319: in _process_results
    for res in results:
datalad/core/local/save.py:222: in __call__
    for s in Status()(
datalad/interface/base.py:873: in _execute_command_
    for r in _process_results(
datalad/interface/utils.py:319: in _process_results
    for res in results:
datalad/core/local/status.py:407: in __call__
    for r in yield_dataset_status(
datalad/core/local/status.py:161: in yield_dataset_status
    status = repo.diffstatus(
datalad/support/gitrepo.py:3110: in diffstatus
    modified = set(
datalad/support/gitrepo.py:3112: in <genexpr>
    for p in self.call_git_items_(
datalad/dataset/gitrepo.py:525: in call_git_items_
    for file_no, line in self._generator_call_git(
datalad/dataset/gitrepo.py:363: in _generator_call_git
    for file_no, content in generator:
<frozen _collections_abc>:356: in __next__
    ???
datalad/runner/nonasyncrunner.py:100: in send
    return self._locked_send(message)
datalad/runner/nonasyncrunner.py:125: in _locked_send
    self._check_result()
datalad/runner/nonasyncrunner.py:96: in _check_result
    self.runner._check_result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <datalad.runner.nonasyncrunner.ThreadedRunner object at 0x7fffee836cc0>

    def _check_result(self):
        if self.exception_on_error is True:
            if self.return_code not in (0, None):
                protocol = self.protocol
                decoded_output = {
                    source: protocol.fd_infos[fileno][1].decode(protocol.encoding)
                    for source, fileno in (
                        ("stdout", protocol.stdout_fileno),
                        ("stderr", protocol.stderr_fileno))
                    if protocol.fd_infos[fileno][1] is not None
                }
>               raise CommandError(
                    cmd=self.cmd,
                    code=self.return_code,
                    stdout=decoded_output.get("stdout", None),
                    stderr=decoded_output.get("stderr", None)
                )
E               datalad.runner.exception.CommandError: CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false ls-files -z -m -d' failed with exitcode 128

datalad/runner/nonasyncrunner.py:298: CommandError
```

</details>

Full logs: [datalad.log](https://github.com/user-attachments/files/17650850/datalad.log)

### Additional context

None

### Notify maintainers

@renesat

### Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.5-arch1-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.7`
 - nixpkgs: `/home/jenni/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/datalad/default.nix
+++ b/pkgs/development/python-modules/datalad/default.nix
@@ -221,6 +221,9 @@ buildPythonPackage rec {
 
     # pbcopy not found
     "test_wtf"
+
+    # CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false ls-files -z -m -d' failed with exitcode 128
+    "test_subsuperdataset_save"
   ];
 
   nativeCheckInputs = [

